components/micro-gateway-core/src/main/ballerina/src/gateway/handler_providers/jwt_auth_provider.bal

```diff
@@ -67,6 +76,25 @@ public type JwtAuthProvider object {
                     var cachedJwt = trap <jwt:CachedJwt>jwtCache.get(jwtToken);
                     //finishing span
                     finishSpan(JWT_CACHE, spanIdCache);
+                    var isJwtTokenCached = self.gatewayCache.retrieveClaimMappingCache(jwtToken);
+                    if (isJwtTokenCached is boolean) {
+                        printDebug(KEY_JWT_AUTH_PROVIDER, "Moddified claims in the cache");
+                     }
+                    else {
+                        printDebug(KEY_JWT_AUTH_PROVIDER, "Moddified claims is not in the cache");
+                        (jwt:JwtPayload | error) payloadBody = getDecodedJWTPayload(jwtToken);
+                        if (payloadBody is jwt:JwtPayload) {
+                            string payloadIssuer = payloadBody["iss"].toString();
+                            if( self.jwtValidatorConfig["issuer"] ==  payloadIssuer){
+                                var result = doMappingContext(invocationContext, self.className, self.claims, self.classLoaded);
+                                jwtToken = authContext?.authToken.toString();
+                                self.gatewayCache.addClaimMappingCache(jwtToken, true);
+                                if (result is auth:Error){
+                                    return result;
+                                }
+                            }
+                        }
+                    }
                     if (cachedJwt is jwt:CachedJwt) {
```

Comment:
Don't we have to use the modified jwt payload to do the rest of the work. I see, here we pass the original payload to subscription filtering (line 128) and so on.